starter-code/routes/index.js

```diff
@@ -1,9 +1,76 @@
-var express = require('express');
-var router = express.Router();
+const express        = require("express");
+const router         = express.Router();
+// User model
+const User           = require("../models/user");
+// Bcrypt to encrypt passwords
+const bcrypt         = require("bcrypt");
+const saltRounds   = 10;
+const ensureLogin = require("connect-ensure-login");
+const passport      = require("passport");
 
-/* GET home page. */
-router.get('/', function(req, res, next) {
-  res.render('index', { title: 'Express' });
+
+router.get('/', ensureNotAuthenticated, (req, res, next) => {
+  res.render('passport/signup');
+});
+
+router.post('/signup', ensureNotAuthenticated, (req, res, next) => {
+
+      const newUser = User({
```

Comment:
You don't need to create the variable and encrypt the password before knowing if you are going to save the user.